### DISCUSSION TITLE
Task Collector 1.0 Super Draft release

### DISCUSSION THREAD

**Discussion by ebullient** [ebullient/obsidian-task-collector#107] [Announcements]
There is a pre-release for a big revision to Task Collector here: 

> Updated / Latest alpha (BRAT)
> 
> https://github.com/ebullient/obsidian-task-collector/releases/tag/1.0.1

The settings are different. What you can do within one note is significantly different, and that means more chances to get some things wrong.

Task Collector is moving away from the fixed notion of "Complete (Canceled)" and "Incomplete" to a much more fluid system that lets you define as many **groups** as you would like.

## Task Groups

Out of the box (especially if you are migrating from earlier versions), you'll have two groups: `default` and `complete`. Each of these groups can be associated with one or more characters. The `default` group is associated with unchecked tasks (a space), and the `complete` group is associated with `x` or `X`. If you like using  a `-` for cancelled tasks, just add that to the list of marks.

You can define text to append, text to remove, task collection behaviors for each group, and you can create as many groups as you would like. 

But.. What is the difference between complete and not? (there is a toggle.. what is the toggle for?)

The complete toggle is mostly just an indicator. 

- In the task selection dialog, marks to complete tasks appear in the top row.
- Other plugins (like dataview) want to know what constitutes a "complete" or done task. That isn't always an `x`.  Task Collector has an API that will return incomplete and complete status marks.

**Comment by Azurewren** [comment]
I updated to 1.0.0 and restarted my Obsidian. Any time I went  into the plugin's options to check out the settings it would only show a blank settings page and my Obsidian would freeze and the only option would be to kill/close the app down.

I rolled back to 0.8.2 and couldn't test any of the settings or whether it worked.

**Reply by ebullient** [reply]
Test Vault Test Vault Test Vault.

Will release another draft today.

**Reply by ebullient** [reply]
New draft released. 

I unzipped the packaged zip file from the release into my plugins directory, copied over my old/existing Task Collector configuration, and everything started. 

You should see something like this (I think obsidian is zapping the version backwards): 

![image](https://user-images.githubusercontent.com/808713/211927320-48a5eddb-07c0-422a-88e9-ba03da3ce53c.png)

**Reply by ebullient** [reply]
I think I'm going to make this a BRAT release, as the manual way is .. fussy.

All set. BRAT release for 1.0.0. That should work better. ;)

**Comment by ebullient** [comment]
1.0.1 released to BRAT: https://github.com/ebullient/obsidian-task-collector/releases/tag/1.0.1
(description updated above).

**Comment by kmaustral** [comment]
When I use this now, it collects all the incomplete tasks along with the completed ones. How can I leave the incomplete ones like before?

**Reply by ebullient** [reply]
Make sure you have 1.0.1, and not 1.0.0 (you may need to tell BRAT to go looking.. )

If you open the task config, you'll see things grouped. Make sure the Area heading for the `default` group is empty.

![image](https://user-images.githubusercontent.com/808713/212205671-b0f13620-8c8a-4a79-9efe-ce8f3bf9e6ed.png)

**Reply by ebullient** [reply]
I will make that placeholder text say `## Example` to make it more obvious that it is a placeholder.

**Comment by kmaustral** [comment]
Thanks. I deleted the space in the first group that it worked as before.
I'm curious to know when it would be useful to collect uncompleted tasks.

**Reply by ebullient** [reply]
I'll publish updated docs shortly.. just note that you aren't constrained to two groups anymore, and I haven't pinned which group any mark is in.

Given the migration, you should have two groups: `default` and `complete`. The space should have been part of the marks in the first (default) group, and the Area heading for the default group should have been empty. The second, `complete` group, should have marks for complete items, and the Area heading set to `## Log` (or whatever you had configured), which should have resulted in the same behavior.

**Reply by ebullient** [reply]
updated docs pushed. ;)

**Comment by RobColes** [comment]
1.0.3 on Windows, binding "Task Collector (TC) Mark with next" is not working for me. Right-click and same command from the context menu works fine. I have tried multiple different keys and there are no conflicts. No errors in the console. Same with previous.
![TC-Keymap](https://user-images.githubusercontent.com/1765065/212630800-1795e280-dc3d-4ad5-aed8-885f74b05c2e.png)

**Comment by RobColes** [comment]
At the moment I can use "Mark task" on an existing task, or on a bullet. Is there any reason that it couldn't be extended to a normal (not checkbox or bullet) line? 

I can use "Toggle Checkbox status" to go from a regular line to a checkbox, but it would be more convenient to be able to have an open check box as the first of the task mark cycle and be able to use the same hotkey mapping to progress, whether the line was already a task/bullet or not.

**Reply by ebullient** [reply]
To make a normal line into a checkbox in one go, you mean?

I can do that with the command in edit mode pretty easily.

**Reply by RobColes** [reply]
Yes, exactly.
